pycontrails/models/cocipgrid/cocip_grid.py

```diff
@@ -448,6 +450,9 @@ def _generate_new_vectors(
 
                 # Convert the 4D grid to a vector
                 vector = source_slice.to_vector()
+                vector.update(longitude=vector["longitude"].astype(self._target_dtype, copy=False))
```

Comment:
So if I interpret this right - you are keeping vector dtypes the same as the met data values?  And the reason to have a different standard here is to reduce memory overhead?

Reply:
Yeah -- you got it right. Sorry this so opaque.

Now, we automatically convert `MetDataset` coords to float64. But I want to be able to run `CocipGrid` with float32 precision for memory and performance reasons. This is a cheap way to let the underlying grid dtype dictate the precision when we convert to vector here.

Definitely confusing!